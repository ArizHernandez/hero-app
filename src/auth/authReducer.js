import { types } from "../types/types";

/* State 
{
  name: 'Marco',
  isLoged: true/false
}
*/ 

export const authReducer = ( state = {}, action ) => {

  switch ( action.type ) {
    case types.login:
        return { 
          ...action.payload, 
          isLoged: true 
        }
    case types.logout:
      return {
        isLoged: false
      }
  
    default:
      return state;
  }
}