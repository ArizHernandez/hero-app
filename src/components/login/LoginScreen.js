import React, { useContext } from 'react'
import { AuthContext } from '../../auth/AuthContext'
import { types } from '../../types/types';
import './style.css'

export const LoginScreen = ({ history }) => {
  const { dispatch } = useContext( AuthContext );

  const handleClick = () => {

    const lastPath = localStorage.getItem('lastPath') || '/';

    dispatch({ 
      type: types.login, 
      payload: { 
        name: 'Ariz' 
      } 
    });

    history.replace( lastPath );
  }

  return (
    <div className="container login__page d-flex justify-content-center align-items-center">

      <div className="login__container rounded text-center text-white mx-auto p-3">
        <h1>Login</h1>
        <hr className="mt-2 mb-4"/>
        <button
          className="btn btn-primary mt-2"
          onClick={ handleClick }
        >
          Login
        </button>
      </div>
    </div>
  )
}
