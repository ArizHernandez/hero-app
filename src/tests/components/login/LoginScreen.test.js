import React from 'react';
import { mount } from "enzyme"
import { AuthContext } from '../../../auth/AuthContext';
import { LoginScreen } from '../../../components/login/LoginScreen';
import { types } from '../../../types/types';

describe('Pruebas en <LoginScreen />', () => {
  
  const context = {
    dispatch: jest.fn()
  };

  const history = {
    replace: jest.fn()
  };

  const wrapper = mount(
    <AuthContext.Provider value={context}>
      <LoginScreen history={history}/>
    </AuthContext.Provider>
  );

  test('Debe de cargar correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  })

  test('Debe de realizar el dispatch y la navegaciÃ³n', () => {
    const handleClick = wrapper.find('button').prop('onClick');

    handleClick();
    expect(context.dispatch).toBeCalledWith({type: types.login, payload: {name: 'Ariz'}});
    expect(history.replace).toBeCalledWith('/')

    localStorage.setItem('lastPath', '/dc');
    handleClick();
    expect(history.replace).toBeCalledWith('/dc');
  })
  
})
