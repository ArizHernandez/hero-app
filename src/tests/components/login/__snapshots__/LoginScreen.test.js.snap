// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> Debe de cargar correctamente 1`] = `
<div
  className="container login__page d-flex justify-content-center align-items-center"
>
  <div
    className="login__container rounded text-center text-white mx-auto p-3"
  >
    <h1>
      Login
    </h1>
    <hr
      className="mt-2 mb-4"
    />
    <button
      className="btn btn-primary mt-2"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
